src/openms/include/OpenMS/CHEMISTRY/ModificationsDB.h

```diff
@@ -80,6 +80,16 @@ namespace OpenMS
       return db_;
     }
 
+    inline static ModificationsDB * getCustomInstance(OpenMS::String unimod_file, OpenMS::String psimod_file)
```

Comment:
well, shouldnt this be local and `db_` for getCustomInstance is a different instance as for getInstance

I am rather worried that two calls to getCustomInstance will return the *previous* `db_` even if now different arguments are given. I see two solutions

- drop the stupid pattern and simply return `new ModificationsDB(unimod_file, psimod_file);` or even `return shared_ptr<ModificationsDB>(new ModificationsDB(unimod_file, psimod_file));`
- keep a static `std::map` around where you store all combinations of the input strings and if you see a new combination of files, create a new `ModificationsDB` otherwise return the one stored in the map.

I dont really see the point of static stuff here, as this funciton will anyways be called from a TOPP tool that has those paths. So I suggest the first option and simply drop the static stuff altogether